Sources/BrowserServicesKit/Autofill/AutofillUserScript.swift

```diff
@@ -19,16 +19,70 @@
 
 import WebKit
 
+#if !os(iOS)
+/// Handles calls from the top Autofill context to the overlay
+public protocol TopOverlayAutofillUserScriptDelegate: AnyObject {
+    /// Provides a size that the overlay should be resized to
+    func requestResizeToSize(width: CGFloat, height: CGFloat)
+}
+
+/// Is used by the top Autofill to reference into the child autofill
+public protocol AutofillMessagingToChildDelegate {
+    /// Represents the last tab host, is used to verify messages origin from and selecting of credentials
+    var lastOpenHost: String? { get }
+    /// Handles filling the credentials back from the top autofill into the child
+    func messageSelectedCredential(_ data: [String: String], _ configType: String)
+    /// Closes the overlay
+    func close()
+}
+
+/// Handles calls from the child Autofill context to the overlay.
+public protocol ChildOverlayAutofillUserScriptDelegate: AnyObject {
+    var view: NSView { get }
+    /// Closes the overlay
+    func autofillCloseOverlay(_ autofillUserScript: AutofillMessagingToChildDelegate?)
```

Comment:
This should pass actual type of the delegate itself, not a different one.

Reply:
`AutofillMessagingToChildDelegate` currently still exists, I'll look at this tomorrow the interface is shared a little more than we'd like but should be simple to fix up.